### ISSUE TITLE
Add Cargo.lock to workspace root

### ISSUE THREAD

**Issue by orhun** [nickgerace/gfold#107]
### Describe the issue.

According to the cargo workspace [documentation](https://doc.rust-lang.org/cargo/reference/workspaces.html), all packages should share a common `Cargo.lock` file which should be present in the workspace root.

> All packages share a common Cargo.lock file which resides in the workspace root.

Also AUR packages will become non-[reproducible](https://reproducible-builds.org/) due to this issue since `--locked` flag causes the build to fail. (because `Cargo.lock` doesn't exist in the workspace root)

```
==> Starting build()...
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /build/gfold/src/gfold-1.0.0/gfld/Cargo.toml
workspace: /build/gfold/src/gfold-1.0.0/Cargo.toml
    Updating crates.io index
error: the lock file /build/gfold/src/gfold-1.0.0/Cargo.lock needs to be updated but --locked was passed to prevent this
If you want to try to generate the lock file without accessing the network, use the --offline flag.
==> ERROR: A failure occurred in build().
```

Solution is simply removing `Cargo.lock` from [.gitignore](https://github.com/nickgerace/gfold/blob/38ebaba01a33e3058676718c3cb0ac86883a4368/.gitignore#L10) and pushing it.

### Is this about `gfold`, `gfld`, or both?

Both, about Cargo workspaces.

### Is this a bug report, feature request, or another type of issue?

Bug report.

### Paste the output of the relevant application(s).

```sh
# to reproduce the build error
git clone "https://aur.archlinux.org/gfold-git"
cd gfold-git/
extra-x86_64-build
```

**Comment by nickgerace** [comment]
Thanks @orhun. I'll get to work on this soon.

**Comment by nickgerace** [comment]
@orhun would it be best for you if I tagged `1.0.1

**Comment by nickgerace** [comment]
@orhun would it be best for you if I tagged `1.0.1` for the AUR and we skipped `1.0.0`,  or would you prefer that we fix `1.0.0`? My objection to the second option would be that `crates.io` already has 1.0.0 out and I would need to republish the crate.

*Side note: apologies, I accidentally closed the issue. Please disregard that.*

**Comment by orhun** [comment]
> @orhun the fix has been added to `main`: [e2738b2](https://github.com/nickgerace/gfold/commit/e2738b27409090a30602ebfa935266d22d3b0f5d)

Thanks!

> if I tagged `1.0.1` and we skipped `1.0.0` in the AUR

That's totally fine.

> _Side note: apologies, I accidentally closed the issue. Please disregard that._

_No problem!_

P.S. You might want to add `--locked` flag to `cargo` commands in your CI just to check if `Cargo.lock` is up-to-date. Otherwise I might have to open up other issue(s) and ping you just for that in the future. ref: bf0e6071ddcfc7a1108e8dc189470772b5251a7d

**Comment by nickgerace** [comment]
Ok great! Added the `--locked` flag back to CI and tagged `1.0.1`.
We should be good to go!
